evaluate_model.py

```diff
@@ -278,6 +305,22 @@ def evaluate_model(transcribe_fn, ds, text_column):
         ds = datasets.load_dataset(dataset_name)[dataset_split]
 
     print(f"Beginning evaluation.")
-    res = evaluate_model(transcribe_fn, ds, ds_text_column)
-
-    print(f"Evaluation done. WER={res.wer}, WIL={res.wil}.")
+    metrics, results_df = evaluate_model(transcribe_fn, ds, ds_text_column)
+
+    print(f"Evaluation done. WER={metrics.wer}, WIL={metrics.wil}.")
+    
+    # Add model and dataset info as columns 
+    results_df['model'] = args.model
+    results_df['dataset'] = dataset_name
+    results_df['dataset_split'] = dataset_split
+    results_df['engine'] = args.engine
+    
+    output_file = args.output if args.output else "evaluation_results"
+    
+    results_df.to_csv(output_file, encoding='utf-8', index=False)
+    
+    with open(f"{output_file}_metrics.pkl", 'wb') as f:
```

Comment:
Why is pickle needed?
You have the CSV anyways, loading it is the same one-liner and it's not lots of data.

Also: https://docs.python.org/3/library/pickle.html